### ISSUE TITLE
"Borrow" all the ideas of Zenkit

### ISSUE THREAD

**Issue by 3ygun** [J3RN/time-tracker#104]
Just watch the video for [https://zenkit.com/](https://zenkit.com/) I want it all...

**Comment by J3RN** [comment]
Alright, here's a breakdown

## Things I'd like
- [ ] Add photos, etc to tasks
- [ ] Kanban view (been on my mental todo list for some time)
- [ ] Attach users to Tasks (many-to-many, #24)
- [ ] Analytics (#60)

## Things I'm skeptical of
- [ ] Invoicing (I'm not terribly skeptical of this, I just wouldn't ever use it).
- [ ] Custom fields + sorting by them. I'm skeptical about the usefulness of that.
- [ ] Cute scheduling interface.
- [ ] "Mind map", aka Topological Ordering.

@3ygun Let me know if I missed anything.

**Comment by 3ygun** [comment]
I agree with you on the **Things I'd like** but on the skeptical side:
- Invoicing >> What do you mean? I'm with you based on my knowledge on invoices but don't know where they were in the product.
- Custom fields >> Agreed, useless for a todo/timesheet app
- 🐤 cute scheduling interface >> would be nice but a lot of work
- Mind Map >> Really? This is one of the coolest features imo (maybe not applicable for a timesheet) but it's not out yet so we'll see

**Comment by J3RN** [comment]
- Invoicing: For all the companies I've worked for, they already have some platform they've been using since time began, and use that for invoicing. The only time I would use that feature would be if I become a freelancer, and that's not looking likely. I'm sure it would be nice for freelancers, if any use my project 😄 
- Cute scheduling interface: From a UX point of view, having to go to a separate page to schedule tasks would just be annoying, unless it's just another way to schedule tasks in addition to the current one. It would, alternately, be nice to have some sort of calendar integration (Google Calendar, perhaps) to visualize when things are due. We'll see.
- Mind Map: I feel like this is basically just a re-implementation of a Kanban board, but I can see some value over that interface, though. The biggest blockers for this feature are A) the existing question of switching between the current interface to Kanban to this one and B) the technical difficulty of implementing this interface in an intuitive way.
